proposals/lip-0052.md

```diff
@@ -679,6 +671,172 @@ else: # chainID == sendingChainID
+            "items": {
+                "type": "object",
+                "required": [
+                    "NFTID",
+                    "owner", 
+                    "lockingModuleID", 
+                    "attributes"
+                ],
+                "properties": {
+                    "NFTID": {
+                        "type": "object",
+                        "fieldNumber": 1,
+                        "required": ["chainID", "collection", "index"],
+                        "properties": {
+                            "chainID": {
+                                "dataType": "uint32",
+                                "fieldNumber": 1
+                            },
+                            "collection": {
+                                "dataType": "uint32",
+                                "fieldNumber": 2
+                            },
+                            "index": {
+                                "dataType": "uint32",
+                                "fieldNumber": 3
+                            }
+                        }
+                    },
+                    "owner": {
+                        "dataType": "bytes",
+                        "fieldNumber": 2
+                    },
+                    "lockingModuleID": {
+                        "dataType": "uint32",
+                        "fieldNumber": 3
+                    },
+                    "attributes": {
+                        "dataType": "bytes",
+                        "fieldNumber": 4
+                    }
+                }
+            }
+        },
+        "collectionSubstore": {
+            "type": "array",
+            "fieldNumber": 2,
+            "items": {
+                "type": "object",
+                "required": ["collection", "nextAvailableIndex"],
+                "properties": {
+                    "collection": {
+                        "dataType": "uint32",
+                        "fieldNumber": 1
+                    },
+                    "nextAvailableIndex": {
+                        "dataType": "uint64",
+                        "fieldNumber": 2
+                    }
+                }
+            }
+        },
+        "availableCollectionSubstore": {
+            "dataType": "uint32",
+            "fieldNumber": 3
+        },
+        "terminatedEscrowSubstore": {
+            "type": "array",
+            "fieldNumber": 4,
+            "items": {
+                "type": "uint32"
+            }
+        }
+    }
+}
+```
+
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the NFT module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisNFTStoreSchema` schema, given above.
+
+* Initial checks:
```

Comment:
Same comments as for the Token module:

I would find it easier for manual validation by a user as well as programmatic validation of the token module if for an array of key-value pairs we always have that the array elements are sorted lexicographically by key.

Reply:
OK, added the requirement here also.

Reply:
I am a bit confused by the changes as they are specified for the collection substore, but seem to refer to the NFT substore. What I meant above:

- `NFT substore`: order by NFT ID or, which may be more convenient, by address and then NFT ID (this would be the key order in the user substore)
- `collectionSubstore`: order by collection
- `terminatedEscrowSubstore`: order the array entries